### ISSUE TITLE
CsvWriter writes binary data to the file (i.e. does not do ascii conversion)

### ISSUE THREAD

**Issue by mark-lazarides** [benhar-dev/tc3-data-logger#2]
here's an example of the output
![image](https://user-images.githubusercontent.com/56538022/98160353-fd366580-1ed5-11eb-952c-3d823c3f28ef.png)

**Comment by AndyP01** [comment]
Agreed, this is known. The spec from the customer that initiated this project had a concern over ultimate file size, and the requirement was simply that they know the data format so it could be post-processed. Therefore keeping the values as raw binary was acceptable.

However, having values written as a string as an option is something a that myself and @benhar-dev have discussed, very briefly.

Open to suggestions :)

**Comment by benhar-dev** [comment]
Currently BOOL, INT default to writing Binary.  You must use AddStringLiteral(BOOL_TO_STRING(true)) for conversions to string which adds a performance hit.  

The next release will default to string.  I am currently performance testing the changes and hope to have this pushed shortly.

**Comment by mark-lazarides** [comment]
Understood on all fronts, but a CSV writer should not write binary. Those options should not be available, even by miss-configurstion.

**Comment by mark-lazarides** [comment]
@benhar-dev , if you create a contract with the consumer, with them specifying the data types for each column during setup of the header row, then you can store all as binary for speed, then batch convert in a separate process later. You could even pass a struct by T_Arg (or just adr and size), memcpy that and cherry pick the ToString later.  Unless of course toString is performant enough to begin with. :-)
